src/main/java/org/opensearch/geospatial/geojson/FeatureFactory.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.geospatial.geojson;
+
+
+import org.opensearch.geospatial.geojson.Feature.FeatureBuilder;
+
+import java.util.Map;
+
+/**
+ * FeatureFactory helps to create {@link Feature} instance based on user input
+ */
+public class FeatureFactory {
+
+    /**
+     * The factory method to create an instance of Feature from input, and, throws {@link IllegalArgumentException} for invalid input.
+     *
+     * @param input the object from where {@link Feature} will be extracted
+     * @return Feature Instance from input
+     * @throws IllegalArgumentException if invalid input is provided
+     */
+    public static Feature create(Map<String, Object> input) throws IllegalArgumentException {
+        String geoJSONType = (String) input.get(Feature.TYPE_KEY);
+        if (geoJSONType == null) {
+            throw new IllegalArgumentException(Feature.TYPE_KEY + " cannot be null");
+        }
+        if (!Feature.TYPE.equalsIgnoreCase(geoJSONType)) {
+            throw new IllegalArgumentException(geoJSONType + " is not supported. Only type " + Feature.TYPE + " is supported");
+        }
+        return extractFeature(input).build();
+    }
+
+    private static FeatureBuilder extractFeature(Map<String, Object> input) throws IllegalArgumentException {
+        Object geometry = input.get(Feature.GEOMETRY_KEY);
+        if (geometry == null) {
+            throw new IllegalArgumentException("key: " + Feature.GEOMETRY_KEY + " cannot be null");
+        }
+        if (!(geometry instanceof Map)) {
+            throw new IllegalArgumentException("key: " + Feature.GEOMETRY_KEY + "is not an instance of type Map");
+        }
+        Map<String, Object> geometryMap = (Map<String, Object>) geometry;
```

Comment:
nit: are we sure this is map of <String, Object>? Although I'm sure how to check that properly, effectively you're getting Map<Object,Object>

Reply:
created new method to convert